### PR TITLE
add support for ember-template-lint v4 [skip ci]

### PR DESCRIPTION
**Pull request by jsturgis**
add support for ember-template-lint v4.

### CODE REVIEW

**Comment by jsturgis** [comment]
@suchitadoshi1987, this is currently blocking us from seeing ember template lint errors in the IDE, any chance you can take a look soon?

**Comment by jsturgis** [comment]
Ping

File: src/template-linter.ts [hunk 1]
```diff
@@ -168,16 +184,10 @@ export default class TemplateLinter {
         return;
       }
 
-      let nodePath = Files.resolveGlobalNodePath();
-
-      // vs-code-online fix (we don't have global path, but it returned)
-      if (!nodePath || !fs.existsSync(nodePath)) {
-        // easy fix case
-        nodePath = 'node_modules';
+      const nodePath = 'node_modules';
```
Comment (chrisrng) [inline src/template-linter.ts:187]: Any reason for removing `resolveGlobalNodePath` here and fallback to `node_modules` instead?

File: src/template-linter.ts [hunk 2]
```diff
@@ -110,7 +122,11 @@ export default class TemplateLinter {
 
     const TemplateLinter = await this.getLinter(project);
 
-    let linter: typeof TemplateLinter | null = null;
+    if (!TemplateLinter) {
+      return;
+    }
```
Comment (chrisrng) [inline src/template-linter.ts:127]: Was this just missed?

File: src/template-linter.ts (hunk 1 above)
Comment (jsturgis) [reply src/template-linter.ts:187]: @chrisrng This change was copied from the ETL4 support PR in the upstream repo [here](https://github.com/lifeart/ember-language-server/pull/347/files#diff-04887b0a2d36f1d0e38ee27bf37b08f2bd41b27f2b526894588dd644301f9338L215)

File: src/template-linter.ts (hunk 2 above)
Comment (jsturgis) [reply src/template-linter.ts:127]: Sorry, I don't follow?

**Comment by jsturgis** [comment]
All changes directly copied or influenced by https://github.com/lifeart/ember-language-server/pull/347.

File: src/template-linter.ts (hunk 1 above)
Comment (chrisrng) [reply src/template-linter.ts:187]: Looks like its because of this https://github.com/ember-template-lint/ember-template-lint/issues/2336

**Review by chrisrng — APPROVED** [review]

File: src/template-linter.ts (hunk 2 above)
Comment (chrisrng) [reply src/template-linter.ts:127]: Looks like this is also upstream

**Pull request merged**